src/app/dataset-block/metadata-block/components/event-details/components/common/base-property/base-property.component.ts

```diff
@@ -0,0 +1,7 @@
+import { EnvVar } from "./../../../../../../../api/kamu.graphql.interface";
+import { Injectable, Input } from "@angular/core";
+
+@Injectable()
+export abstract class BasePropertyComponent {
+    @Input() public data: unknown | string[] | EnvVar[];
```

Comment:
I don't see sense in having this union input. Instead, put a well-typed input in each subclas. No benefits from calling different things with the name `data`.